extensions/wikia/Follow/FollowHelper.class.php

```diff
@@ -100,17 +100,15 @@ public static function emailNotification($childTitle, $list, $namespace, $user,
 			}
 		}
 
-
-		foreach ($watchers as $key => $value) {
-			$enotif = new EmailNotification();
-			$title = Title::makeTitle( $namespace, $key );
-			$enotif->notifyOnPageChange( $user, $title,
-				$now,
-				$message,
-				0,
-				0,
-				$action,
-				array('notisnull' => 1, 'childTitle' => $childTitle) );
+		/**
+		 * Selecting all watching users took to long
+		 * and was causing timeouts. It's been moved to a task.
+		 * @see CE-1239 by adamk@wikia-inc.com
+		 */
+		if ( !empty( $watchers ) ) {
+			$oTask = new FollowEmailTask();
+			$oTask->call( 'emailFollowNotifications', $watchers, $user->getId(), $namespace, $message, $action, $childTitle->getDBkey() );
```

Comment:
I would recommend passing in the title with `$oTask->title(Title::makeTitle($namespace, $key))`. This allows you to reference the title as `$this->title` when the task is executing. We added this because there were some cases where the encoding for the title params was getting messed up somehow as it passed from php -> rabbitmq -> python -> php. Using `->title()` fixes those edge cases.